### ISSUE TITLE
Clover tests fail in feature/generic_kernel

### ISSUE THREAD

**Issue by bjoo** [lattice/quda#1089]
The clover unit tests fail in `feature/generic_kernel` (and cascade also to `feature/generic-kernel-hotfixes` and `feature/hip-compile-fixes` )

Symptom: 
```
Running tests...
Test project /home/users/coe0071/QUDA-Hipify/build_attempt/nvbuild
    Start 1: blas_test_parity_wilson
1/8 Test #1: blas_test_parity_wilson ..........   Passed   35.04 sec
    Start 2: blas_test_full_wilson
2/8 Test #2: blas_test_full_wilson ............   Passed   36.10 sec
    Start 3: blas_interface_test
3/8 Test #3: blas_interface_test ..............   Passed    2.22 sec
    Start 4: dslash_wilson-policytune
4/8 Test #4: dslash_wilson-policytune .........   Passed    6.24 sec
    Start 5: dslash_clover-sym-policytune
5/8 Test #5: dslash_clover-sym-policytune .....***Failed    2.15 sec
    Start 6: dslash_clover-asym-policytune
6/8 Test #6: dslash_clover-asym-policytune ....***Failed    2.19 sec
    Start 7: gauge_alg_single
7/8 Test #7: gauge_alg_single .................   Passed    8.86 sec
    Start 8: gauge_alg_double
8/8 Test #8: gauge_alg_double .................   Passed    8.79 sec
```

Closer look with `ctest -V -I 5,6` gives
```
5: [ RUN      ] QUDA/DslashTest.verify/half_r18_partition0
5: Kappa = 0.12195122 Mass = 0.10000000
5: clover.TrLog()[0]=-nan, clover.TrLog()[1]=-4.134776e-01
5: ERROR: Clover trlog has returned -nan, likey due to the clover matrix being singular. (rank 0, host node001, clover_invert.cu:38 in apply())
5:        last kernel called was (name=N4quda12CloverInvertIsEE,volume=2x4x6x8,aux=trlog=true,twist=false)
```
and
```
6: Kappa = 0.12195122 Mass = 0.10000000
6: clover.TrLog()[0]=-nan, clover.TrLog()[1]=-4.134776e-01
6: ERROR: Clover trlog has returned -nan, likey due to the clover matrix being singular. (rank 0, host node001, clover_invert.cu:38 in apply())
6:        last kernel called was (name=N4quda12CloverInvertIsEE,volume=2x4x6x8,aux=trlog=true,twist=false)
```

**Comment by bjoo** [comment]
Test system used V100, CUDA 10.2, GCC 8.3.1

**Comment by cpviolator** [comment]
Try this: https://github.com/lattice/quda/tree/hotfix/GK_clover_tr_log

I think it's because we're using a random matrix.

**Comment by cpviolator** [comment]
Fixed with 049191be64ac3955df94c5e7fddaf742c87ab5fc

**Comment by weinbe2** [comment]
@cpviolator @bjoo as an FYI the issue is almost certainly because it's a random matrix---I ran into a similar issue when I was testing the gamma5 work with the clover operator, as the random clover isn't gamma5 Hermitian either. One way around this is using `--compute-clover true`, which we should add in some fashion to the `ctest`.
